chainer/training/updater.py

```diff
@@ -250,9 +250,11 @@ def __init__(self, iterator, optimizer, converter=convert.concat_examples,
             models = {'main': optimizer.target}
             for name in names:
                 model = optimizer.target.copy()
-                model.to_gpu(devices[name])
+                if devices[name] is not None and devices[name] >= 0:
```

Comment:
Thanks for valuable info! This case, I basically followed the implementation of `to_device` in `concat_examples`. I thought there may be `None` values in `devices` dictionary, and we should handle them correctly. `names` are from the keys of `devices`, so default values are never used.